[Bug]: Matrix Channel feature compilation failure

### Affected component

channel

### Severity

S1 - workflow blocked

### Current behavior

Compiling ZeroClaw with `cargo` and feature flag for `matrix-sdk` fails. See logs section for console output.

Note: there is an issue with https://github.com/rust-lang/rust/issues/152942 (`rustc 1.94`) compiling the `matrix-sdk` crate  that is unrelated to this one.

### Expected behavior

Compilation should complete successfully.

### Steps to reproduce

```bash
1. `$ cargo build --features channel-matrix`
2. Observe Clippy warnings and compilation error.
```

### Impact

_No response_

### Logs / stack traces

```text
warning: unused import: `InReplyTo`
 --> src/channels/matrix.rs:8:40
  |
8 |         events::relation::{Annotation, InReplyTo, Thread},
  |                                        ^^^^^^^^^
  |
  = note: `#[warn(unused_imports)]` (part of `#[warn(unused)]`) on by default

error[E0433]: failed to resolve: use of undeclared type `Relation`
   --> src/channels/matrix.rs:572:43
    |
572 |                 content.relates_to = Some(Relation::Thread(Thread::plain(
    |                                           ^^^^^^^^ use of undeclared type `Relation`

warning: unused variable: `matrix_skip_context`
    --> src/channels/mod.rs:2955:5
     |
2955 |     matrix_skip_context: &str,
     |     ^^^^^^^^^^^^^^^^^^^ help: if this is intentional, prefix it with an underscore: `_matrix_skip_context`
     |
     = note: `#[warn(unused_variables)]` (part of `#[warn(unused)]`) on by default
```

### ZeroClaw version

5d1543100d485e5db0bc2fe264d7230b14f1ebf5

### Rust version

rustc 1.93

### Operating system

Linux debian-s-1vcpu-1gb-sfo3-01 6.12.74+deb13+1-amd64 #1 SMP PREEMPT_DYNAMIC Debian 6.12.74-2 (2026-03-08) x86_64 GNU/Linux

### Regression?

Unknown

### Pre-flight checks

- [x] I reproduced this on the latest master branch or latest release.
- [x] I redacted secrets, tokens, and personal data from all submitted content.

--- a/src/channels/matrix.rs
+++ b/src/channels/matrix.rs
@@ -5,9 +5,9 @@ use matrix_sdk::{
     config::SyncSettings,
     ruma::{
         events::reaction::ReactionEventContent,
-        events::relation::{Annotation, InReplyTo, Thread},
+        events::relation::{Annotation, Thread},
         events::room::message::{
-            MessageType, OriginalSyncRoomMessageEvent, RoomMessageEventContent,
+            MessageType, OriginalSyncRoomMessageEvent, Relation, RoomMessageEventContent,
         },
         events::room::MediaSource,
         OwnedEventId, OwnedRoomId, OwnedUserId,
@@ -540,12 +540,7 @@ impl Channel for MatrixChannel {
     async fn send(&self, message: &SendMessage) -> anyhow::Result<()> {
         let client = self.matrix_client().await?;
         let target_room_id = if message.recipient.contains("||") {
-            message
-                .recipient
-                .splitn(2, "||")
-                .nth(1)
-                .unwrap()
-                .to_string()
+            message.recipient.split_once("||").unwrap().1.to_string()
         } else {
             self.target_room_id().await?
         };
@@ -589,8 +584,7 @@ impl Channel for MatrixChannel {
             let tts_text = message
                 .content
                 .replace("**", "")
-                .replace("*", "")
-                .replace("`", "")
+                .replace(['*', '`'], "")
                 .replace("# ", "");
 
             let tts_ok = tokio::process::Command::new("edge-tts")
@@ -703,7 +697,7 @@ impl Channel for MatrixChannel {
 
         client.add_event_handler(move |event: OriginalSyncRoomMessageEvent, room: Room| {
             let tx = tx_handler.clone();
-            let target_room = target_room_for_handler.clone();
+            let _target_room = target_room_for_handler.clone();
             let my_user_id = my_user_id_for_handler.clone();
             let allowed_users = allowed_users_for_handler.clone();
             let dedupe = Arc::clone(&dedupe_for_handler);
@@ -736,7 +730,7 @@ impl Channel for MatrixChannel {
                                 format!("{}/_matrix/client/v1/media/download/{}", homeserver, rest);
                             Some((url, name.to_string()))
                         }
-                        _ => None,
+                        MediaSource::Encrypted(_) => None,
                     }
                 };
 
@@ -782,7 +776,7 @@ impl Channel for MatrixChannel {
                     {
                         Ok(resp) if resp.status().is_success() => match resp.bytes().await {
                             Ok(bytes) => match tokio::fs::write(&dest, &bytes).await {
-                                Ok(_) => format!("{} — saved to {}", body, dest.display()),
+                                Ok(()) => format!("{} — saved to {}", body, dest.display()),
                                 Err(_) => format!("{} — failed to write to disk", body),
                             },
                             Err(_) => format!("{} — download failed", body),

--- a/src/channels/mod.rs
+++ b/src/channels/mod.rs
@@ -2990,6 +2990,7 @@ fn collect_configured_channels(
     config: &Config,
     matrix_skip_context: &str,
 ) -> Vec<ConfiguredChannel> {
+    let _ = matrix_skip_context;
     let mut channels = Vec::new();
 
     if let Some(ref tg) = config.channels_config.telegram {

